File: src/struct_frame/boilerplate/rust/frame_headers.rs
// Frame Headers - Start byte patterns and header configurations (Rust)
// Mirrors frame_headers.h from C boilerplate

/// Header type enumeration
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum HeaderType {
    None,
    Tiny,
    Basic,
}

/// Constants
pub const BASIC_START_BYTE: u8 = 0x90;
pub const PAYLOAD_TYPE_BASE: u8 = 0x70;

/// Header configuration
#[derive(Debug, Clone, Copy)]
pub struct HeaderConfig {
    pub header_type: HeaderType,
    pub start_byte1: u8,
    pub num_start_bytes: u8,
    pub encodes_payload_type: bool,
}

/// Get the start byte for a Tiny frame with given payload type
pub fn get_tiny_start_byte(payload_type_value: u8) -> u8 {
    PAYLOAD_TYPE_BASE + payload_type_value
}

/// Get the second start byte for a Basic frame with given payload type
pub fn get_basic_second_start_byte(payload_type_value: u8) -> u8 {
    PAYLOAD_TYPE_BASE + payload_type_value
}

// Pre-defined header configurations

/// None header - no start bytes
pub const HEADER_NONE_CONFIG: HeaderConfig = HeaderConfig {
    header_type: HeaderType::None,
    start_byte1: 0,
    num_start_bytes: 0,
    encodes_payload_type: false,
};

/// Tiny header - 1 start byte [0x70+PayloadType]
pub const HEADER_TINY_CONFIG: HeaderConfig = HeaderConfig {
    header_type: HeaderType::Tiny,
    start_byte1: 0, // Dynamic - depends on payload type
    num_start_bytes: 1,
    encodes_payload_type: true,
};

/// Basic header - 2 start bytes [0x90] [0x70+PayloadType]
pub const HEADER_BASIC_CONFIG: HeaderConfig = HeaderConfig {
    header_type: HeaderType::Basic,
    start_byte1: BASIC_START_BYTE,
    num_start_bytes: 2,
    encodes_payload_type: true,
};

File: src/struct_frame/boilerplate/js/frame-headers.js
"use strict";
/**
 * Frame Headers - Start byte patterns and header configurations (TypeScript)
 * Header types define start byte patterns and header-specific parsing
 *
 * This file mirrors the C++ frame_headers.hpp structure.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HEADER_MAVLINK_V2_CONFIG = exports.HEADER_MAVLINK_V1_CONFIG = exports.HEADER_UBX_CONFIG = exports.HEADER_BASIC_CONFIG = exports.HEADER_TINY_CONFIG = exports.HEADER_NONE_CONFIG = exports.MAX_PAYLOAD_TYPE = exports.MAVLINK_V2_STX = exports.MAVLINK_V1_STX = exports.UBX_SYNC2 = exports.UBX_SYNC1 = exports.PAYLOAD_TYPE_BASE = exports.BASIC_START_BYTE = exports.HeaderType = void 0;
/** Header type enumeration */
var HeaderType;
(function (HeaderType) {
    HeaderType[HeaderType["NONE"] = 0] = "NONE";
    HeaderType[HeaderType["TINY"] = 1] = "TINY";
    HeaderType[HeaderType["BASIC"] = 2] = "BASIC";
    HeaderType[HeaderType["UBX"] = 3] = "UBX";
    HeaderType[HeaderType["MAVLINK_V1"] = 4] = "MAVLINK_V1";
    HeaderType[HeaderType["MAVLINK_V2"] = 5] = "MAVLINK_V2"; // 1 start byte [0xFD]
})(HeaderType || (exports.HeaderType = HeaderType = {}));
/** Constants used across headers */
exports.BASIC_START_BYTE = 0x90;
exports.PAYLOAD_TYPE_BASE = 0x70; // Payload type encoded as 0x70 + payload_type
exports.UBX_SYNC1 = 0xB5;
exports.UBX_SYNC2 = 0x62;
exports.MAVLINK_V1_STX = 0xFE;
exports.MAVLINK_V2_STX = 0xFD;
exports.MAX_PAYLOAD_TYPE = 8;
/** Pre-defined header configurations */
exports.HEADER_NONE_CONFIG = {
    headerType: HeaderType.NONE,
    startByte1: 0,
    startByte2: 0,
    numStartBytes: 0,
    encodesPayloadType: false,
};
exports.HEADER_TINY_CONFIG = {
    headerType: HeaderType.TINY,
    startByte1: 0, // Dynamic - 0x70 + payload_type
    startByte2: 0,
    numStartBytes: 1,
    encodesPayloadType: true,
};
exports.HEADER_BASIC_CONFIG = {
    headerType: HeaderType.BASIC,
    startByte1: exports.BASIC_START_BYTE, // 0x90, then dynamic 0x70 + payload_type
    startByte2: 0,
    numStartBytes: 2,
    encodesPayloadType: true,
};
exports.HEADER_UBX_CONFIG = {
    headerType: HeaderType.UBX,
    startByte1: exports.UBX_SYNC1, // 0xB5
    startByte2: exports.UBX_SYNC2, // 0x62
    numStartBytes: 2,
    encodesPayloadType: false,
};
exports.HEADER_MAVLINK_V1_CONFIG = {
    headerType: HeaderType.MAVLINK_V1,
    startByte1: exports.MAVLINK_V1_STX, // 0xFE
    startByte2: 0,
    numStartBytes: 1,
    encodesPayloadType: false,
};
exports.HEADER_MAVLINK_V2_CONFIG = {
    headerType: HeaderType.MAVLINK_V2,
    startByte1: exports.MAVLINK_V2_STX, // 0xFD
    startByte2: 0,
    numStartBytes: 1,
    encodesPayloadType: false,
};
